packages/zoe/src/contracts/attestation/attestation.js

```diff
@@ -0,0 +1,158 @@
+// @ts-check
+
+import { AmountMath } from '@agoric/ertp';
+import { E } from '@agoric/eventual-send';
+import { Far } from '@agoric/marshal';
+
+import '../../../exported';
+import { setupAttestation as setupExpiringAttestation } from './expiring/expiringNFT';
+import { setupAttestation as setupReturnableAttestation } from './returnable/returnableNFT';
+import { makeStoredTime } from './storedTime';
+import { max } from './helpers';
+import { assertPrerequisites } from './prerequisites';
+import { makeGetAttMaker } from './attMaker';
+
+/**
+ * @type {ContractStartFn}
+ */
+const start = async zcf => {
+  const {
+    brands: { Underlying: underlyingBrand },
+    authority, // The authority used to confirm that the underlying assets are escrowed appropriately
+    expiringAttName,
+    returnableAttName,
+  } = zcf.getTerms();
+
+  const { assetKind: underlyingAssetKind } = await E(
+    underlyingBrand,
+  ).getDisplayInfo();
+  // AWAIT ///
+
+  const storedTime = makeStoredTime();
+  const empty = AmountMath.makeEmpty(underlyingBrand, underlyingAssetKind);
+
+  const expiringAttManagerP = setupExpiringAttestation(
+    expiringAttName, // e.g. 'BldAttGov'
+    empty,
+    zcf,
+  );
+  const returnableAttManagerP = setupReturnableAttestation(
+    returnableAttName, // e.g. 'BldAttLoc'
+    empty,
+    zcf,
+  );
+  const [expiringAttManager, returnableAttManager] = await Promise.all([
+    expiringAttManagerP,
+    returnableAttManagerP,
+  ]);
+  // AWAIT ///
+
+  /** @type {GetLienAmount} */
+  const getLienAmount = (address, currentTime) => {
+    const expiringLienAmount = expiringAttManager.getLienAmount(
+      address,
+      currentTime,
+    );
+    const returnableLienAmount = returnableAttManager.getLienAmount(address);
+    return max(expiringLienAmount, returnableLienAmount);
+  };
+
+  /** @type {GetLiened} */
+  const getLiened = (addresses, currentTime) => {
```

Comment:
Only need to pass in a single address at a time. Need to pass in a `Brand` for the denom.